qiskit_ibm_runtime/api/auth.py

```diff
@@ -40,9 +45,24 @@ def __call__(self, r: PreparedRequest) -> PreparedRequest:
         r.headers.update(self.get_headers())
         return r
 
+    def _get_access_token(self) -> str:
+        """Return IBM Cloud access token."""
+        try:
+            url = CLOUD_IAM_URL
+            if "test" in self.url:
+                url = STAGING_CLOUD_IAM_URL
+            headers = {"Content-Type": "application/x-www-form-urlencoded"}
+            data = f"grant_type=urn:ibm:params:oauth:grant-type:apikey&apikey={self.api_key}"
+            response = post(url, headers=headers, data=data, timeout=10).json()
+            return response["access_token"]
+        except Exception as ex:  # pylint: disable=broad-except
+            raise IBMNotAuthorizedError(
+                "Unable to retrieve IBM Cloud access token. Please check your credentials"
+            ) from ex
+
```

Comment:
We only need this one endpoint - should this logic be in it's own Cloud auth client?